packages/graphiql/src/components/DocExplorer/TypeDoc.tsx

```diff
@@ -5,207 +5,176 @@
+  if (type instanceof GraphQLEnumType) {
+    const values = type.getValues();
+    valuesDef = (
+      <div className="doc-category">
+        <div className="doc-category-title">values</div>
+        {values
+          .filter(value => Boolean(!value.deprecationReason))
+          .map(value => (
+            <EnumValue key={value.name} value={value} />
+          ))}
+      </div>
+    );
 
-    let valuesDef: ReactNode;
-    let deprecatedValuesDef: ReactNode;
-    if (type instanceof GraphQLEnumType) {
-      const values = type.getValues();
-      valuesDef = (
+    const deprecatedValues = values.filter(value =>
+      Boolean(value.deprecationReason),
+    );
+    if (deprecatedValues.length > 0) {
+      deprecatedValuesDef = (
         <div className="doc-category">
-          <div className="doc-category-title">values</div>
-          {values
-            .filter(value => Boolean(!value.deprecationReason))
-            .map(value => (
+          <div className="doc-category-title">deprecated values</div>
+          {!showDeprecated ? (
+            <button
+              className="show-btn"
+              onClick={() => {
+                setShowDeprecated(true);
+              }}>
+              Show deprecated values...
+            </button>
+          ) : (
+            deprecatedValues.map(value => (
               <EnumValue key={value.name} value={value} />
-            ))}
+            ))
+          )}
         </div>
       );
-
-      const deprecatedValues = values.filter(value =>
-        Boolean(value.deprecationReason),
-      );
-      if (deprecatedValues.length > 0) {
-        deprecatedValuesDef = (
-          <div className="doc-category">
-            <div className="doc-category-title">deprecated values</div>
-            {!this.state.showDeprecated ? (
-              <button className="show-btn" onClick={this.handleShowDeprecated}>
-                Show deprecated values...
-              </button>
-            ) : (
-              deprecatedValues.map(value => (
-                <EnumValue key={value.name} value={value} />
-              ))
-            )}
-          </div>
-        );
-      }
     }
-
-    return (
-      <div>
-        <MarkdownContent
-          className="doc-type-description"
-          markdown={
-            ('description' in type && type.description) || 'No Description'
-          }
-        />
-        {type instanceof GraphQLObjectType && typesDef}
-        {fieldsDef}
-        {deprecatedFieldsDef}
-        {valuesDef}
-        {deprecatedValuesDef}
-        {!(type instanceof GraphQLObjectType) && typesDef}
-      </div>
-    );
   }
 
-  handleShowDeprecated = () => this.setState({ showDeprecated: true });
+  return (
+    <div>
+      <MarkdownContent
+        className="doc-type-description"
+        markdown={
+          ('description' in type && type.description) || 'No Description'
+        }
+      />
+      {type instanceof GraphQLObjectType && typesDef}
```

Comment:
I'd probably use the helpers from graphql-js here, because those will also work across multiple GraphQL versions https://github.com/graphql/graphql-js/pull/3617